lib/inputstreamhelper/__init__.py

```diff
@@ -146,7 +158,15 @@ def _widevine_path(cls):
 
         if cls._ia_cdm_path():
             widevine_path = os.path.join(cls._ia_cdm_path(), widevine_cdm_filename)
-            from xbmcvfs import exists
+
+            # Support vendor-supplied Widevine CDM
+            vendor_widevine_path = '{0}_vendor{1}'.format(*os.path.splitext(widevine_path))
+            if exists(vendor_widevine_path):
+                if exists(widevine_path):
+                    if not samefile(vendor_widevine_path, widevine_path):
+                        hardlink(vendor_widevine_path, widevine_path)
+                else:
+                    hardlink(vendor_widevine_path, widevine_path)
```

Comment:
I think it made more sense if `_widevine_path()` only did what its name suggests, i.e. return the path to the Widevine lib.
If these lines were moved to `_has_vendor_widevine()` and a check for `_has_vendor_widevine()` was added to `_has_widevine()` I think it would still work and in my opinion would be more logical.

Reply:
@horstle I don't disagree, but there's more duplicate code if I split it off :disappointed: I implemented now what you asked for.